src/controllers/UserNewsController.js

```diff
@@ -0,0 +1,155 @@
+const axios = require("axios");
+const userNewsPreferences = require("../data/userNewsPreferences.json");
+const userReadNews = require("../data/userReadNews.json");
+const userFavouriteNews = require("../data/userFavouriteNews.json");
+const { updateUserNewsPreferences } = require("../helpers");
+const Validation = require("../utilities/Validation");
+const redisClient = require("../cache/redis");
+const { newsURL } = require("../config");
+
+const parsedUserPreferences = JSON.parse(JSON.stringify(userNewsPreferences));
+
+const fetchNews = async (req, res) => {
+  const userId = req.id;
+  const userPreference = parsedUserPreferences[userId];
+
+  if (userPreference) {
```

Comment:
checking the failure cases first will reduce these complex if-else blocks and reduce code complexity